src/libraries/System.Private.CoreLib/src/System/Text/Utf8StringComparer.cs

```diff
@@ -57,8 +57,8 @@ public static Utf8StringComparer FromComparison(StringComparison comparisonType)
 
         private sealed class CultureAwareComparer : Utf8StringComparer
         {
-            internal static readonly CultureAwareComparer Invariant = new CultureAwareComparer(CompareInfo.Invariant, CompareOptions.None);
-            internal static readonly CultureAwareComparer InvariantIgnoreCase = new CultureAwareComparer(CompareInfo.Invariant, CompareOptions.IgnoreCase);
+            internal static readonly CultureAwareComparer Invariant = new CultureAwareComparer(CultureInfo.InvariantCulture.CompareInfo, CompareOptions.None);
```

Comment:
Avoid bouncing through `CultureInfo.InvariantCulture` when in-box.

Reply:
Even though this is cached statically? At runtime it shouldn't cause any perf differences, should it?

I can easily add more `#if`s here. I just thought it wouldn't be a problem for this case.

Reply:
Yeah, even for things which are statically cached it matters to an extent. We want to involve as few unnecessary code paths as possible in the static initialization of these types. It puts less pressure on the JIT and helps optimize AOT scenarios.